subsys/bluetooth/services/fast_pair/fp_registration_data.c

```diff
@@ -24,14 +25,17 @@ static const uint8_t fp_magic[] = {0xFA, 0x57, 0xFA, 0x57};
 #define FP_OFFSET(_prev_item_offset, _prev_item_len) \
 	(_prev_item_offset + ROUND_UP(_prev_item_len, FP_DATA_ALIGN))
 
-#define FP_MAGIC_HEADER_OFF	 FP_DATA_START
-#define FP_MODEL_ID_OFF		 FP_OFFSET(FP_MAGIC_HEADER_OFF, FP_MAGIC_SIZE)
+#define FP_MAGIC_OFF		 FP_DATA_START
+#define FP_MODEL_ID_OFF		 FP_OFFSET(FP_MAGIC_OFF, FP_MAGIC_SIZE)
 #define FP_ANTI_SPOOFING_KEY_OFF FP_OFFSET(FP_MODEL_ID_OFF, FP_REG_DATA_MODEL_ID_LEN)
-#define FP_MAGIC_TRAILER_OFF	 FP_OFFSET(FP_ANTI_SPOOFING_KEY_OFF, \
+#define FP_HASH_OFF		 FP_OFFSET(FP_ANTI_SPOOFING_KEY_OFF, \
 					   FP_REG_DATA_ANTI_SPOOFING_PRIV_KEY_LEN)
+#define FP_END_OFF		 FP_OFFSET(FP_HASH_OFF, FP_CRYPTO_SHA256_HASH_LEN)
 
-BUILD_ASSERT(FP_OFFSET(FP_MAGIC_TRAILER_OFF, FP_MAGIC_SIZE) <= FP_PARTITION_SIZE,
-	     "Fast Pair registration data partition is too small");
+#define FP_DATA_SIZE		 (FP_END_OFF - FP_DATA_START)
+#define FP_OFFSET_TO_DATA_IDX(_offset) (_offset - FP_DATA_START)
```

Comment:
Please add brackets:

```suggestion
#define FP_OFFSET_TO_DATA_IDX(_offset) ((_offset) - FP_DATA_START)
```